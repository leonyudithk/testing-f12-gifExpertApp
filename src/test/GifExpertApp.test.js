import React from 'react'
import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp"

describe('Pruebas al Componente GitExpertApp', ()=>{

    test('Debe mostrarse correctamente la categoria', ()=>{
      
        //Arreglar- preparar mi mesa de trabajo, las declaraciones varibles y las importaciones
        const ca = ['Hola']
        const wrapper = shallow(<GifExpertApp defaultCategories = {ca}/>)

        //Actuar- voy a realizar mi test prueba unitaria

        expect(wrapper).toMatchSnapshot()

        //afirmarv- este es solo observar los resultados y analizar a
    })


    test("Mostrar una lista categorias", ()=>{
        //Arreglar
        const categoria = ['Marvel', 'flores', 'osos', 'gatos', 'perros', '##@@@']

        const wrapper = shallow( <GifExpertApp defaultCategories={categoria}/>)

        //actuar
        expect(wrapper).toMatchSnapshot()

    })
    test("debe mostrar todas las categorias", ()=>{
        //Arreglar
        const categorias = ['Dragon Ball', 'MC' ,'##11@@@']

        const wrapper = shallow( <GifExpertApp defaultCategories={categorias}/>)

        //actuar
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categorias.length)

    })


    
   
})