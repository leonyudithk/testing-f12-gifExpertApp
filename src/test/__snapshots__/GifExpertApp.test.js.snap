// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al Componente GitExpertApp Debe mostrarse correctamente la categoria 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hola"
      key="Hola"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al Componente GitExpertApp Mostrar una lista categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Marvel"
      key="Marvel"
    />
    <GifGrid
      category="flores"
      key="flores"
    />
    <GifGrid
      category="osos"
      key="osos"
    />
    <GifGrid
      category="gatos"
      key="gatos"
    />
    <GifGrid
      category="perros"
      key="perros"
    />
    <GifGrid
      category="##@@@"
      key="##@@@"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al Componente GitExpertApp debe mostrar todas las categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="MC"
      key="MC"
    />
    <GifGrid
      category="##11@@@"
      key="##11@@@"
    />
  </ol>
</Fragment>
`;
